### ISSUE TITLE
celebA dataset smile issue 

### ISSUE THREAD

**Issue by geopapyrus** [HasnainRaz/Fast-AgingGAN#5]
Hi!

 I tried to train on the celeb A dataset to make photo go from non-smile to smile and it didn't work.
 Do you have any advice?
 

thank you

**Comment by HasnainRaz** [comment]
It's difficult to say without reviewing your code. There could be multiple reasons.

**Comment by geopapyrus** [comment]
Can I email you what I have ? You can send me an email at geopapyrus@vanossi.com.br and I’ll send you the details! Thanks!!!

**Comment by geopapyrus** [comment]
Hi!

 Here's my code https://drive.google.com/file/d/18gAl6Vpzg_5aH0gPFRNg6lBrWozG_xSv/view?usp=sharing
 I made a 2nd experiment now cropping the face a bit more zoomed on the face and I also changed weights of loss components.
  It seems that network is training. slowly, but training
![image](https://user-images.githubusercontent.com/4298488/88083393-f9fab200-cb50-11ea-85eb-684aeb35c142.png)

  See this pic of original vs smiled after 50 epochs. 
 Maybe it needs like 100 epochs?

 Any feedback would be greatly appreciated!
thank you

**Comment by HasnainRaz** [comment]
Hey, sorry for the delay in getting back to you.

Thanks for sharing your code, I'll have a look and get back to you

**Comment by aomv** [comment]
Hi!

 Did you take a look? It seems smile effect gets very tiny.

thank you

**Comment by HasnainRaz** [comment]
Looking at your images, I think the biggest issue is the data itself. Most examples in pos and neg overlap (close mouth smiles are common in both datasets). This probably makes it difficult to optimize the loss function by focusing on the smiles, as they are not discriminative between the two domains. You may try sorting out close mouth smile from both datasets or at least having open mouth smiles only in the pos datasets.

If you don't want to try this out, maybe look at methods likehttps://github.com/taki0112/UGATIT where the adaptive normalization and the classifiers might be able to better focus on the smiling aspect. They will probably not be very fast to run though.
